helper-tools/prepare-jamf-policy.sh

```diff
@@ -87,45 +87,57 @@ _RESULT
 if [ "$plist_type" -lt 1100 ]; then
     # Build a package of installer
     if [ ! -x /usr/bin/pkgbuild ]; then exit 0; fi
-    read -r -p "Would you need a package archive of $( basename "$OSInstaller" )? [y/n]: " ANS
+    read -r -p "Would you need a package archive of $( /usr/bin/basename "$OSInstaller" )? [y/n]: " ANS
     if [ "$ANS" != y ]; then exit 0 ;fi
 
-    echo "Ok, building package archive file of $( basename "$OSInstaller"). Wait few minutes."
+    echo "Ok, building package archive file of $( /usr/bin/basename "$OSInstaller" ). Wait few minutes."
 
     workdir="$(/usr/bin/mktemp -d)"
     PKGID="macOSUpgrade.helper-tools.pkgbuild"
-    PKGFILE="${HOME}/Downloads/$(basename "$OSInstaller" ).${osversion}.pkg"
+    PKGFILE="${HOME}/Downloads/$( /usr/bin/basename "$OSInstaller" ).${osversion}.pkg"
     /bin/mkdir -p "$workdir/root/Applications"
     /bin/cp -a "${OSInstaller%/}" "$workdir/root/Applications"
 
     if [ -f "$PKGFILE" ]; then
         /bin/mv "${PKGFILE}" "${PKGFILE%.pkg}.previous.$(uuidgen).pkg"
     fi
 
-    if /usr/bin/pkgbuild --identifier "$PKGID" --root "${workdir}/root" "$PKGFILE" > "/tmp/pkgbuid.$( date +%F_%H%M%S ).log" ; then
+    if /usr/bin/pkgbuild --identifier "$PKGID" --root "${workdir}/root" "$PKGFILE" > "/tmp/pkgbuid.$( /bin/date +%F_%H%M%S ).log" ; then
         echo "Done. $PKGFILE"
     else
         echo "FAILED."
     fi
 else
     # https://www.jamf.com/jamf-nation/discussions/37294/package-big-sur-installer-with-composer-issue
-    read -r -p "Would you need a DMG file of $( basename "$OSInstaller" )? [y/n]: " ANS
+    read -r -p "Would you need a DMG file of $( /usr/bin/basename "$OSInstaller" )? [y/n]: " ANS
     if [ "$ANS" != y ]; then exit 0 ;fi
 
-    echo "Ok, building DMG file of $( basename "$OSInstaller"). Wait few minutes."
+    echo "Ok, building DMG file of $( /usr/bin/basename "$OSInstaller" ). Wait few minutes."
 
     workdir="$(/usr/bin/mktemp -d)"
     temp_dmg="${workdir}/osinstaller.dmg"
-    dist_dmg="${HOME}/Downloads/$(basename "$OSInstaller" ).${osversion}.dmg"
-    sizeOfInstaller="$( /usr/bin/du -sm "$OSInstaller" | awk '{print $1}' )"
+    dist_dmg="${HOME}/Downloads/$( /usr/bin/basename "$OSInstaller" ).${osversion}.dmg"
+    sizeOfInstaller="$( /usr/bin/du -sm "$OSInstaller" | /usr/bin/awk '{print $1}' )"
     volumename="macOSInstaller"
+    extra_size=512
+    filesystem='APFS'
 
-    /usr/bin/hdiutil create -size $(( sizeOfInstaller + 256 ))m -volname "$volumename" "$temp_dmg" -fs APFS > /dev/null
-    devfile="$( /usr/bin/hdiutil attach -readwrite -nobrowse "$temp_dmg" | awk '$NF == "GUID_partition_scheme" {print $1}' )"
+    if [ "$( /usr/bin/sw_vers -productVersion | /usr/bin/awk -F. '{ print ($1 * 10 ** 2 +  $2 )}' )" -lt 1015 ]; then
+        filesystem='JHFS+'
```

Comment:
In mojave, we have to use Mac OS Extended Journaled and that's an error.
Although this project is also guaranteed to work on High Sierra. Does it work?

Reply:
@kenchan0130 @taniguti thank you for trying to keep "legacy" support. I was running into this for a PR I am working on submitting this morning. What are your thoughts on using the n-2 methodology? That would have us guaranteeing support for the last 3 major releases. We wouldn't remove support necessarily for older versions, but we also wouldn't regularly test against them. Thoughts?

Reply:
> What are your thoughts on using the n-2 methodology?

I agree with this policy.
However, I just checked the project readme as it was described as supported.

Reply:
Yes, it has been and I think we should try, but if it’s gonna cause a lot of additional work then probably not worth it. 

Let’s maintain support for the moment unless we’re hitting a major issue. I’ll reach to to some folks and try and gauge what kind of issues we may run into as far as push back.

Reply:
Yes. I think we must focus recent 3 major releases. Supporting "legacy" itself must make additional tasks and make our code even more complex (and none feels happy).

Thanks @kc9wwh  for checking the script. The output looks good.

Reply:
Certainly! Let's just be sure to link to the previous version on the Readme when we drop support for versions so if someone needs to deploy an older version they can go that commit and grab the older version....does that make sense?